lib/tidewave/mcp/connection.ex

```diff
@@ -196,7 +205,9 @@ defmodule Tidewave.MCP.Connection do
       end
     catch
       kind, reason ->
-        {:error, "Failed to call tool: #{Exception.format(kind, reason, __STACKTRACE__)}"}
+        {:reply,
+         {:error, "Failed to call tool: #{Exception.format(kind, reason, __STACKTRACE__)}"},
+         state}
```

Comment:
This is the main fix for the UUID error. It will still complain when passing a string, but the model can work with the error response now instead of waiting for a reply that never comes.